server/src/recipe/handler/recipe.handler.ts

```diff
@@ -19,19 +20,63 @@ export class RecipeHandler {
     };
   }
 
+  private _resolveCategory = (recipe: Recipe): Promise<Recipe> => {
+    return this._db.recipeCategories.getById(recipe.categoryId)
+      .then((category: RecipeCategory) => {
+        recipe.category = category;
+        return recipe;
+      });
+  }
+
+  private _resolveAllCategories = (recipes: Recipe[]): Promise<Recipe[]> => {
```

Comment:
```ts
private _resolveAllCategories = async (recipes: Recipe[]): Promise<Recipe[]> => {
    return await Promise.all(recipes.map(async (recipe: Recipe) => {
      return await recipe.categoryId
        ? this._resolveCategory(recipe)
        : Promise.resolve(recipe);
    }));
  }
```
async await with map. looks simpler!?

Reply:
Yes this looks better. But in this case you will wait for each _resolveCategor Promise to resolve. So the requests for each category are performed sequentially. The promise only approach will perform the requests in parallel. I will see that I can use the best of both, ok?

Reply:
With request I mean the db query for a category